File: YoutubeClone/DTO/UserSummary.cs
namespace YoutubeClone.DTO
{
    public class UserSummary
    {
        public Guid Id { get; set; }
        public string UserName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string GivenName { get; set; } = string.Empty;
        public string Surname { get; set; } = string.Empty;
        public string? ProfilePictureUrl { get; set; }
        public string PhoneNumber { get; set; } = string.Empty;
        public DateTime Created { get; set; }
    }
}

File: YoutubeClone/ClientApp/src/api/models/user-summary.ts
export interface UserSummary {
    id: string;
    userName: string;
    email: string;
    givenName: string;
    surname: string;
    profilePictureUrl: string;
    phoneNumber: string;
    dateCreated: Date;
}

export function mapJsonToUserSummary(json: any): UserSummary {
    return {
        id: json.id,
        userName: json.userName,
        email: json.email,
        givenName: json.givenName,
        surname: json.surname,
        profilePictureUrl: json.profilePictureUrl,
        phoneNumber: json.phoneNumber,
        dateCreated: json.dateCreated
    }
}